### PR TITLE
Add CI github action

### PR DESCRIPTION
**Pull request by kyledurand**
Co-Authored-By: aveline <aveline@users.noreply.github.com>
Co-Authored-By: Alex Page <19199063+alex-page@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  test:
+    name: Build, type check, lint, and test 💚
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.0'
+
+      - name: Get yarn cache directory path
```
[the remark this answers is not in the document]
Comment (alex-page) [reply .github/workflows/ci.yml:15]: Oh yeah we should replace yarn cache with that lerna example. That caches the yarn node modules in the sub directories as well.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  test:
+    name: Build, type check, lint, and test 💚
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.0'
+
+      - name: restore lerna
```
Comment (samrose3) [inline .github/workflows/ci.yml:15]: I think this should be its own `step`. And maybe a bit nit picky, but could we call the step names Sentence case, "Restore Lerna cache"

This is a guess based on how Charka has their CI [workflow organized](https://github.com/chakra-ui/chakra-ui/blob/main/.github/workflows/pr.yml#L19)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+      - run: yarn --frozen-lockfile
+      - run: yarn build
```
Comment (samrose3) [inline .github/workflows/ci.yml:24]: Could we add a step title for each of these commands? Similar to [this](https://github.com/seek-oss/vanilla-extract/blob/master/.github/workflows/validate.yml#L26)

File: .github/workflows/size-limit.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ jobs:
     env:
       CI_JOB_NUMBER: 1
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
```
Comment (samrose3) [inline .github/workflows/size-limit.yml:16]: Nice update 🚀

**Review by samrose3 — CHANGES_REQUESTED** [review]

**Review by samrose3 — APPROVED** [review]
Way to take Action 💥 🦸  lgtm

**Pull request merged**